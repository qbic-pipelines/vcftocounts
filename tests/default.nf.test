nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    tag "pipeline"
    tag "pipeline_vcftocounts"
    tag "cpu"

    test("-profile test") {

        when {
            params {
                outdir    = "$outputDir"
                input     = "${projectDir}/tests/input.csv"
                removeIDs = true
                rename    = true
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            // vcf_files: All files in ${params.outdir}
            def vcf_files  = getAllFilesFromDir(params.outdir, include: ['**/*.vcf{,.gz}'])
            def csv_files = getAllFilesFromDir(params.outdir, include: ['**/*.csv'])
            assert workflow.success
            assertAll(
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/vcftocounts_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path,
                    // All vcf files
                    vcf_files.collect{ file -> file.name + ":md5," + path(file.path).vcf.variantsMD5 },
                    csv_files.collect{ file -> path(file.path).readLines()[0..10] }
                ).match() }
            )
        }
    }

    test("-profile test - stub") {

        options "-stub"

        when {
            params {
                outdir = "$outputDir"
                input  = "${projectDir}/tests/input.csv"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            // vcf_files: All files in ${params.outdir}
            def vcf_files  = getAllFilesFromDir(params.outdir, include: ['**/*.vcf{,.gz}'])
            assert workflow.success
            assertAll(
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/vcftocounts_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path,
                    // All vcf files
                    vcf_files
                ).match() }
            )
        }
    }
}
