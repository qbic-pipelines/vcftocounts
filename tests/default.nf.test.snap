{
    "-profile test - stub": {
        "content": [
            6,
            {
                "BCFTOOLS_ANNOTATE": {
                    "bcftools": 1.2
                },
                "BCFTOOLS_REHEADER": {
                    "bcftools": 1.2
                },
                "CREATE_SAMPLE_FILE": {
                    "create_sample_file": "1.0.0"
                },
                "TABIX_TABIX": {
                    "tabix": 1.2
                },
                "VCF2COUNTS": {
                    "vcf2counts.R": "1.0.0"
                },
                "Workflow": {
                    "qbic-pipelines/vcftocounts": "v2.0.2dev"
                }
            },
            [
                "bcftools",
                "bcftools/annotate",
                "bcftools/annotate/chr22.IDremoved.vcf.gz",
                "bcftools/annotate/chr22.IDremoved.vcf.gz.tbi",
                "bcftools/annotate/versions.yml",
                "bcftools/reheader",
                "bcftools/reheader/chr22.reheader.vcf.gz",
                "bcftools/reheader/chr22.reheader.vcf.gz.tbi",
                "bcftools/reheader/samplefiles",
                "bcftools/reheader/samplefiles/chr22.txt",
                "bcftools/reheader/samplefiles/versions.yml",
                "bcftools/reheader/versions.yml",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_plots",
                "multiqc/multiqc_report.html",
                "pipeline_info",
                "pipeline_info/vcftocounts_software_mqc_versions.yml",
                "tabix",
                "tabix/NA12878.chr22.1X.vcf.gz.csi",
                "tabix/NA12878.chr22.1X.vcf.gz.tbi",
                "vcf2counts",
                "vcf2counts/chr22.csv"
            ],
            [
                "versions.yml:md5,74b0a613672a9941d2bb2a62c18406bb",
                "chr22.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "versions.yml:md5,4afe7b519f82d5d8c539950934477c1a",
                "versions.yml:md5,e516c8ae57ef27b417031a3d10715b00",
                "chr22.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
            ],
            [
                "chr22.IDremoved.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "chr22.reheader.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-05-14T08:57:56.081242"
    },
    "-profile test": {
        "content": [
            6,
            {
                "BCFTOOLS_ANNOTATE": {
                    "bcftools": 1.2
                },
                "BCFTOOLS_REHEADER": {
                    "bcftools": 1.2
                },
                "CREATE_SAMPLE_FILE": {
                    "create_sample_file": "1.0.0"
                },
                "TABIX_TABIX": {
                    "tabix": 1.2
                },
                "VCF2COUNTS": {
                    "vcf2counts.R": "1.0.0"
                },
                "Workflow": {
                    "qbic-pipelines/vcftocounts": "v2.0.2dev"
                }
            },
            [
                "bcftools",
                "bcftools/annotate",
                "bcftools/annotate/chr22.IDremoved.vcf.gz",
                "bcftools/annotate/chr22.IDremoved.vcf.gz.tbi",
                "bcftools/annotate/chr22.reheader.vcf.gz.csi",
                "bcftools/annotate/versions.yml",
                "bcftools/reheader",
                "bcftools/reheader/chr22.reheader.vcf.gz",
                "bcftools/reheader/chr22.reheader.vcf.gz.tbi",
                "bcftools/reheader/samplefiles",
                "bcftools/reheader/samplefiles/chr22.txt",
                "bcftools/reheader/samplefiles/versions.yml",
                "bcftools/reheader/versions.yml",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_citations.txt",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_software_versions.txt",
                "multiqc/multiqc_data/multiqc_sources.txt",
                "multiqc/multiqc_report.html",
                "pipeline_info",
                "pipeline_info/vcftocounts_software_mqc_versions.yml",
                "tabix",
                "tabix/NA12878.chr22.1X.vcf.gz.tbi",
                "vcf2counts",
                "vcf2counts/chr22.csv"
            ],
            [
                "versions.yml:md5,74b0a613672a9941d2bb2a62c18406bb",
                "chr22.txt:md5,5ae68a67b70976ee95342a7451cb5af1",
                "versions.yml:md5,4afe7b519f82d5d8c539950934477c1a",
                "versions.yml:md5,e516c8ae57ef27b417031a3d10715b00",
                "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f",
                "chr22.csv:md5,e7980f307dbbb17f000b8c36dc47fdfb"
            ],
            [
                "chr22.IDremoved.vcf.gz:md5,5e7f2837f7f33de49ced557ef2fc5bfa",
                "chr22.reheader.vcf.gz:md5,11558863c409ca2d6278e699cd5fde1a"
            ],
            [
                [
                    "\"\",\"chr22\"",
                    "\"chr22:16570065_A/G\",0",
                    "\"chr22:16570067_A/C\",0",
                    "\"chr22:16570176_C/A\",0",
                    "\"chr22:16570211_T/C\",0",
                    "\"chr22:16570267_T/C\",0",
                    "\"chr22:16570325_T/C\",0",
                    "\"chr22:16570434_A/G\",0",
                    "\"chr22:16570441_G/A\",0",
                    "\"chr22:16570473_A/G\",0",
                    "\"chr22:16570478_G/T\",0"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-05-14T08:57:35.234908"
    }
}